dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -0,0 +1,165 @@
+) : RumMonitor {
+
+    private val activeView = mutableMapOf<Any, RumEvent>()
+    private val activeResources = mutableMapOf<Any, RumEvent>()
+
+    // region RumMonitor
+
+    override fun addUserAction(
+        action: String,
+        attributes: Map<String, Any?>
+    ) {
+        val eventData = RumEventData.UserAction(action)
+        val event = RumEvent(
+            GlobalRumMonitor.getRumContext(),
+            timeProvider.getServerTimestamp(),
+            eventData,
+            attributes
+        )
+        writer.write(event)
+    }
+
+    override fun startView(
+        key: Any,
+        name: String,
+        attributes: Map<String, Any?>
+    ) {
+        val activeViewId = UUID.randomUUID()
+        GlobalRumMonitor.updateViewId(activeViewId)
+        val eventData = RumEventData.View(name, System.nanoTime())
+        val event = RumEvent(
+            GlobalRumMonitor.getRumContext(),
+            timeProvider.getServerTimestamp(),
+            eventData,
+            attributes
+        )
+        activeView[key] = event
+    }
+
+    override fun stopView(
+        key: Any,
+        className: String,
+        attributes: Map<String, Any?>
+    ) {
+        val startedEvent = activeView.remove(key)
+        val startedEventData = startedEvent?.eventData as? RumEventData.View
+
+        if (startedEvent == null) {
+            devLogger.w(
+                "Unable to stop view with key <$key> and class <$className>. " +
+                    "This can mean that the view was not started or already stopped."
+            )
+        } else if (startedEventData == null) {
+            devLogger.e(
+                "Unable to stop view with key <$key> and class <$className>. " +
+                    "The related data was inconsistent."
+            )
+        } else {
+            val updatedEvent = startedEvent.copy(
+                eventData = startedEventData.copy(
+                    durationNanoSeconds = System.nanoTime() - startedEventData.durationNanoSeconds,
+                    version = startedEventData.version + 1
+                ),
+                attributes = startedEvent.attributes.toMutableMap().apply {
+                    putAll(attributes)
+                }
+            )
+            writer.write(updatedEvent)
+        }
+        GlobalRumMonitor.updateViewId(null)
+    }
+
+    override fun startResource(key: Any, url: String, attributes: Map<String, Any?>) {
+        val eventData = RumEventData.Resource(
+            RumEventData.Resource.Kind.OTHER,
+            url,
+            System.nanoTime()
+        )
+        val event = RumEvent(
+            GlobalRumMonitor.getRumContext(),
+            timeProvider.getServerTimestamp(),
+            eventData,
+            attributes
+        )
+        activeResources[key] = event
+    }
+
+    override fun stopResource(key: Any, mimeType: String?, attributes: Map<String, Any?>) {
+        val startedEvent = activeResources.remove(key)
```

Comment:
startedResource

Reply:
the type is RumEvent, so it still works as is :/